doc/specs/stdlib_strings.md

```diff
@@ -538,3 +538,69 @@ program demo_count
 
 end program demo_count
 ```
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### `to_string`
+
+#### Description
+
+Format or transfer a `integer/real/complex/logical` scalar as a string.  
+Input a wrong `format` that cause the internal-IO to fail, the result value is a string of `[*]`.
+
+#### Syntax
+
+`string = [[stdlib_strings(module):to_string(interface)]] (value [, format])`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Pure function.
+
+#### Argument
+
+- `value`: Shall be an `integer/real/complex/logical` scalar.
+  This is an `intent(in)` argument.
+- `format`: Shall be a `character(len=*)` scalar like `'(F6.2)'`.
```

Comment:
@zoziha @awvwgk Suggestion for docs improvement, to go together with https://github.com/fortran-lang/stdlib/pull/444#discussion_r693108150.

```suggestion
- `format`: Shall be a `character(len=*)` scalar that contains the edit descriptor to format `value` into a string,
  for example `'(F6.2)'` or `'(f6.2)'`. `to_string` will automatically enclose `format` in a set of parentheses,
  so passing `F6.2` or `f6.2` as `format` is possible as well.
```